src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/TokenCertificateService.java

```diff
@@ -101,6 +101,27 @@ public CertificateInfo getCertificateInfo(String hash) throws CertificateNotFoun
         return certificateInfo;
     }
 
+    /**
+     * Find an existing cert by it's hash and import
+     * @param hash cert hash of an existing cert
+     * @return CertificateType
+     * @throws CertificateNotFoundException
+     * @throws InvalidCertificateException other general import failure
+     * @throws GlobalConfService.GlobalConfOutdatedException
+     * @throws KeyNotFoundException
+     * @throws CertificateAlreadyExistsException
+     * @throws WrongCertificateUsageException
+     * @throws ClientNotFoundException
+     * @throws CsrNotFoundException
+     */
+    public CertificateType importExistingCertificate(String hash) throws CertificateNotFoundException,
```

Comment:
In both cases it all comes down into iterating `currentTokens` in `TokenManager`. In the old implementation the iteration is done in the Ruby layer by first fetching the token by it's id (`TokenManager#getTokenInfo`) and in the new implementation we are utilizing the already implemented method in `TokenManager` class (`TokenManager#getCertificateInfoForCertHash`). 

So in other words the source of the information is the same in both cases.

Reply:
I was thinking about a case where two different tokens (or keys) would contain a cert with same hash. In the old implementation cert would be imported from the token and key that the user selected. In the new implementation cert would be picked based on which one happens to be iterated first.

But if the hashes as same, then the cert should be the same, and it should not matter which one gets imported. I can't think of a practical use case where this would cause a problem.

Reply:
> But if the hashes as same, then the cert should be the same

^ This. I can't think of a real concrete problem in here either.